cf-custom-resources/lib/custom-domain-app-runner.js

```diff
@@ -95,7 +95,10 @@ exports.handler = async function (event, context) {
             }),
         });
         appRunnerClient = new AWS.AppRunner();
-        appHostedZoneID = await domainHostedZoneID(appDNSName);
+        appHostedZoneID = props.RootHostedZoneId
+        if (!appHostedZoneID){
```

Comment:
Why do we pass `RootHostedZoneId` to custom resource, instead of looping through what we get from the `listHostedZonesByName` call and find the hosted zone that's not private?

Reply:
Since we have 6 custom resources using the same `listHostedZonesByName`. I need to replicate the same exact logic for PublicHostedZoneId which we have in the [`route53`](https://github.com/aws/copilot-cli/blob/e6730742f5f232e84b46b0abdc3f95d0bcc71ae1/internal/pkg/aws/route53/route53.go#L52) api call. 
I need to replicate same logic in all the six custom resources. I am not sure which is better way whether to replicate same logic in all the custom resources or pass these hostedzone id as parameters in custom resources.
So I went with second approach.